src/Magnum/Trade/MeshData.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "MeshData.h"
+
+namespace Magnum { namespace Trade {
+
+MeshData::MeshData(Containers::Array<char>&& indexData, Containers::Array<char>&& vertexData, const MeshIndexData indices, Containers::Array<MeshAttributeData>&& attributes, const void* const importerState) noexcept: _indexData{std::move(indexData)}, _vertexData{std::move(vertexData)}, _indices{indices}, _attributes{std::move(attributes)}, _importerState{importerState} {
+    /** @todo assert that index data are divisible by type size, same for other types */
+    /** @todo assert that there's at least one position? */
+    /** @todo assert that all arrays have the same amount of elemenc */
+}
+
+MeshData::~MeshData() = default;
+
+MeshData::MeshData(MeshData&&) noexcept = default;
+
+MeshData& MeshData::operator=(MeshData&&) noexcept = default;
+
+UnsignedInt MeshData::indexCount() const {
+    switch(_indices.type) {
+        case MeshIndexType::UnsignedByte: return _indices.data.size();
+        case MeshIndexType::UnsignedShort: return _indices.data.size()/2;
+        case MeshIndexType::UnsignedInt: return _indices.data.size()/4;
```

Comment:
How about `return _indices.data.size()/indexSize(_indices.type);` ? Maybe that can be used for the `indices()` function also and removes those switches...

If the MeshIndexType enum coincidentally would be it's own length, i.e. Byte = 1, Short = 2, Int = 4, this would be only a cast even

Reply:
Oh! Cool optimization. Though, a bit dangerous and undetectable when people pass garbage there :)